### PR TITLE
community: make LocalAIEmbeddings comaptible with openai>1.0

### PR DESCRIPTION
**Pull request by liunux4odoo**
I run the `bge-large` embedding model by xinference locally, but the latest `LocalAIEmbeddings can only work with `openai<1.0` , not comaptible with the latest `openai`. 

I upgrade the `LocalAIEmbeddings` referencing to `ChatOpenAI` to make it working with `openai>1.0`.

### CODE REVIEW

**Comment by eahova** [comment]
these changes work great for me using localai embeddings with news openai - thanks!

submitted a pr to your fork for the lint errors

**Comment by bdqfork** [comment]
hope we can merge as soon as possible. I have encountered the same problem.

**Comment by eahova** [comment]
I moved this to #18799 and got it to pass the CI tests

**Comment by liunux4odoo** [comment]
@eahova I have merge your pr for lint fixing.

**Comment by imClumsyPanda** [comment]
This PR helps a lot! Plz review asap  @baskaryan @eyurtsev @hwchase17

File: libs/community/langchain_community/embeddings/localai.py
Comment (baskaryan) [inline libs/community/langchain_community/embeddings/localai.py:1]: could we make this compatible with both openai v0 and v1, the way langchain_community.embedding.OpenAIEmbeddings is? or if that's too hard we should at least add an explicit version check of the openai sdk and raise an informative error if it's less than 1

File: libs/community/langchain_community/embeddings/localai.py [hunk 1]
```diff
@@ -42,11 +52,11 @@ def _create_retry_decorator(embeddings: LocalAIEmbeddings) -> Callable[[Any], An
...
-            retry_if_exception_type(openai.error.Timeout)
-            | retry_if_exception_type(openai.error.APIError)
-            | retry_if_exception_type(openai.error.APIConnectionError)
-            | retry_if_exception_type(openai.error.RateLimitError)
-            | retry_if_exception_type(openai.error.ServiceUnavailableError)
+            retry_if_exception_type(openai.Timeout)
+            | retry_if_exception_type(openai.APIError)
+            | retry_if_exception_type(openai.APIConnectionError)
+            | retry_if_exception_type(openai.RateLimitError)
+            | retry_if_exception_type(openai.InternalServerError)
```
Comment (baskaryan) [inline libs/community/langchain_community/embeddings/localai.py:59]: don't think this is v0 compatible

File: libs/community/langchain_community/embeddings/localai.py [hunk 2]
```diff
@@ -264,11 +291,15 @@ def _embedding_func(self, text: str, *, engine: str) -> List[float]:
...
-        )["data"][0]["embedding"]
+        return (
+            embed_with_retry(
+                self,
+                input=[text],
+                **self._invocation_params,
+            )
+            .data[0]
+            .embedding
+        )
```
Comment (baskaryan) [inline libs/community/langchain_community/embeddings/localai.py:302]: don't think this is v0 compatible

File: libs/community/langchain_community/embeddings/localai.py
Comment (liunux4odoo) [reply libs/community/langchain_community/embeddings/localai.py:1]: surely it has been done.

File: libs/community/langchain_community/embeddings/localai.py (hunk 2 above)
Comment (baskaryan) [reply libs/community/langchain_community/embeddings/localai.py:338]: bump

Comment (baskaryan) [reply libs/community/langchain_community/embeddings/localai.py:338]: same below

**Comment by ChristianSch** [comment]
Hey @baskaryan any updates here? Anything you need help with? I'm interested in getting this released asap

**Comment by liunux4odoo** [comment]
I would like to close this PR because latest langchain-openai works well with locally served embeddings by xinference.

**Pull request closed without merging**